### PR TITLE
Other structure for Data Encryption Service

### PR DESCRIPTION
**Pull request by andrzejsamson**
Changed the structure of Data Encryption Service. Removed redundant Model class and separated GUI from logic.

### CODE REVIEW

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionController.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.capgemini.mrchecker.test.core.utils.encryption.CryptParams;
+
+public interface DataEncryptionController {
+	
+	void onEncrypt(CryptParams cryptParams);
+	
+	void onDecrypt(CryptParams cryptParams);
+	
+	void onKey(JTextField keyTextField);
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionController.java]: This should not be a part of DataEncryptionController

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import org.jasypt.exceptions.EncryptionOperationNotPossibleException;
+
+import com.capgemini.mrchecker.test.core.base.encryption.providers.DataEncryptionService;
+import com.capgemini.mrchecker.test.core.exceptions.BFSecureModuleException;
+import com.capgemini.mrchecker.test.core.utils.encryption.CryptParams;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IView;
+
+public class DataEncryptionControllerImp implements DataEncryptionController {
+	private final IView view;
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java]: Change IView to DataEncryptionView

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	
+	@Override
+	public void onEncrypt(CryptParams cryptParams) {
+		try {
+			DataEncryptionService.init(new ByteArrayInputStream(cryptParams.getSecret()
+					.getBytes()));
+			view.setEncryptionFieldValue(DataEncryptionService.getInstance()
+					.encrypt(cryptParams.getText()));
+		} catch (BFSecureModuleException e) {
+			showMessageDialog(null, "Could not encrypt because:\n" + e.getMessage());
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java]: Do not show dialog here. Throw unchecked exception and handle it in the view.

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			view.setEncryptionFieldValue("");
+		} finally {
+			DataEncryptionService.delInstance();
+		}
+	}
+	
+	@Override
+	public void onDecrypt(CryptParams cryptParams) {
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java]: onEncrypt and onDecrypt have duplicates. Please refactor. You can use functional interface to implement that.

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		} catch (EncryptionOperationNotPossibleException e) {
+			showMessageDialog(null, "Could not decrypt because:\nWrong key for given encrypted text");
+			view.setEncryptionFieldValue("");
+		} finally {
+			DataEncryptionService.delInstance();
+		}
+	}
+	
+	@Override
+	public void onKey(JTextField keyTextField) {
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java]: Let the view control opening files.

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/view/DataEncryptionGUI.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+	private JTextField decryptionResultTextField;
+	private JButton decryptButton;
+	private JButton chooseDecryptionKeyButton;
+	private JTextField decryptionKeyTextField;
+	private JLabel decryptionKeyLabel;
+	private JLabel decryptionLabel;
+	private JLabel decryptionResultLabel;
+	
+	public DataEncryptionGUI() {
+		dataEncryptionController = new DataEncryptionControllerImp(this);
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/view/DataEncryptionGUI.java]: This cannot be implemented like that. We need another way to instantiate and pass DataEncryptionController implementation.

File: mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/view/IView.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.capgemini.mrchecker.test.core.utils.encryption.view;
+
+import javax.swing.*;
+
+public interface IView {
+	void setEncryptionFieldValue(String text);
+	
+	void setDecryptionFieldValue(String text);
+	
+	void setKeyFieldValue(String text, JTextField keyTextField);
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/view/IView.java]: Let's talk about that

**Review by michalbabiarz — CHANGES_REQUESTED** [review]

**Comment by andrzejsamson** [comment]
TODO:
-tests for business logic from app
-instruction how to run and use the app

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+import com.capgemini.mrchecker.test.core.tags.IntegrationTest;
+import com.capgemini.mrchecker.test.core.utils.encryption.controller.DataEncryptionController;
+import com.capgemini.mrchecker.test.core.utils.encryption.controller.IDataEncryptionController;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.DataEncryptionGUI;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IDataEncryptionView;
+
+@IntegrationTest
+@ResourceLock(value = "SingleThread")
+public class DataEncryptionGUITest {
+	private static final IDataEncryptionController dataEncryptionController = new DataEncryptionController();
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: Using a concrete class. A mock should be used.

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+		dataEncryptionGUIFixture.textBox("encryptionKeyTextField")
+				.enterText(cryptParams.getSecret());
+		dataEncryptionGUIFixture.textBox("encryptionTextField")
+				.enterText(cryptParams.getText());
+		dataEncryptionGUIFixture.button("encryptButton")
+				.click();
+		String result = dataEncryptionGUIFixture.textBox("encryptionResultTextField")
+				.text();
+		
+		assertThat(result, startsWith("ENC("));
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: "ENC(" is coupled with DataEncrytpionController. Use mock instead and define return value.

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+	}
+	
+	@Test
+	public void shouldSetSecretFromFileAndEncrypt() throws IOException {
+		dataEncryptionGUIFixture.textBox("encryptionTextField")
+				.enterText(value);
+		dataEncryptionGUIFixture.button("chooseEncryptionKeyButton")
+				.click();
+		dataEncryptionGUIFixture.fileChooser()
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: Do not use file. Give some explicit value.

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+		dataEncryptionGUIFixture.button("encryptButton")
+				.click();
+		dataEncryptionGUIFixture.dialog()
+				.button()
+				.click();
+		
+		String result = dataEncryptionGUIFixture.textBox("encryptionResultTextField")
+				.text();
+		
+		assertThat(result, is(equalTo("")));
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: Code duplication?

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+				.selectTab("Decryption");
+		dataEncryptionGUIFixture.textBox("decryptionTextField")
+				.enterText(cipherForSecretDataFile);
+		dataEncryptionGUIFixture.button("chooseDecryptionKeyButton")
+				.click();
+		dataEncryptionGUIFixture.fileChooser()
+				.setCurrentDirectory(new File(System.getProperty("user.dir")))
+				.selectFile(new File(System.getProperty("user.dir") + "/src/resources/secretData"))
+				.approve();
+		dataEncryptionGUIFixture.button("decryptButton")
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: Do not use a file.

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+				.click();
+		dataEncryptionGUIFixture.dialog()
+				.button()
+				.click();
+		
+		String result = dataEncryptionGUIFixture.textBox("decryptionResultTextField")
+				.text();
+		
+		assertThat(result, is(equalTo("")));
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: Code duplication?

**Review by michalbabiarz — CHANGES_REQUESTED** [review]

File: mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.ResourceLock;
+
+import com.capgemini.mrchecker.test.core.tags.IntegrationTest;
+import com.capgemini.mrchecker.test.core.utils.encryption.controller.DataEncryptionController;
+import com.capgemini.mrchecker.test.core.utils.encryption.controller.IDataEncryptionController;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.DataEncryptionGUI;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IDataEncryptionView;
+
+@IntegrationTest
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java]: Can be a UnitTest

**Review by michalbabiarz — CHANGES_REQUESTED** [review]

**Pull request merged**